libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -88,108 +142,123 @@ static bool _packetMatch( const IotLink_t * pSubscriptionLink,
 
 /*-----------------------------------------------------------*/
 
-static bool _topicMatch( const IotLink_t * pSubscriptionLink,
-                         void * pMatch )
+static bool _matchTerminalWildcards( const char * pTopicFilter,
+                                     uint16_t nameIndex,
+                                     uint16_t filterIndex,
+                                     uint16_t topicNameLength,
+                                     uint16_t topicFilterLength )
 {
-    bool status = false;
-    uint16_t nameIndex = 0, filterIndex = 0;
+    bool status = false, terminalChar = false;
 
-    /* Because this function is called from a container function, the given link
-     * must never be NULL. */
-    IotMqtt_Assert( pSubscriptionLink != NULL );
+    /* Determine if the terminal character is reached for both topic name and topic
```

Comment:
Again, terminal seems to be an ambiguous term in this context, but maybe that's because I don't have the knowledge of the MQTT specification.

Reply:
I'll update this comment to be clearer.  "terminal" just refers to the last character in the string.